cheri-c-programming.tex

```diff
@@ -896,7 +896,21 @@ \subsubsection{Out-of-bounds pointers}
 
 \note{I feel like this section wants a reference to CHERI Concentrate?}{nwf}
 
-The capability compression model exploits redundancy between the pointer and
+ISO C permits pointers to go only one byte beyond their original
+allocation, but real code often \psnote{not sure how strong to be
+  there} constructs transient pointer values which are further out of
+bounds.  To support this, capabilities can hold a range of
+out-of-bounds addresses while retaining a valid tag.  Dereferencing an
+out-of-bounds pointer will raise a hardware exception (see
+\Cref{sec:faults}).  However, an out-of-bounds pointer can be
+dereferenced once it has been brought back in bounds by adjusting the
+address or suppling a suitable offset in the dereference.  This means
```

Comment:
Maybe it's slightly easier to understand for readers if we explicitly state when bounds checks are performed?
E.g. "CHERI-enabled hardware performs bounds checks on pointer use (i.e. dereference) and not pointer manipulation."

Reply:
Agreed.  I've taken a stab.